From 5e6fd3899d034ec833459bd4bffd5e9abbe581fb Mon Sep 17 00:00:00 2001
From: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
Date: Mon, 22 Apr 2024 14:32:54 +0800
Subject: [PATCH 2/2] phy: realtek: enable master mode configuration for
 RTL8211F

The PHY has an issue when operating in gigabit slave mode.
It not able to transmit or receive any packet in the slave
mode. Thus, this patch enable the master mode configuration
of the phy.

Signed-off-by: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
---
 drivers/net/phy/realtek.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/phy/realtek.c b/drivers/net/phy/realtek.c
index a5f1df18a65d..5d21b0195ec9 100644
--- a/drivers/net/phy/realtek.c
+++ b/drivers/net/phy/realtek.c
@@ -240,6 +240,10 @@ static int rtl8211f_config_init(struct phy_device *phydev)
 			val_rxdly ? "enabled" : "disabled");
 	}
 
+	/* RTL8211F has an issue when operating in Gigabit slave mode */
+	phy_set_bits(phydev, MII_CTRL1000,
+		CTL1000_ENABLE_MASTER | CTL1000_AS_MASTER);
+
 	return 0;
 }
 
-- 
2.17.1

